File: python/leetcode/container_with_most_water.py
class Solution:
    # @return an integer

    def maxArea(self, height):
        if not height or len(height) < 2:
            return 0

        best = float('-inf')
        left, right = 0, len(height) - 1

        while left < right:
            if height[left] > height[right]:
                best = max(best, (right - left) * height[right])
                right -= 1
            else:
                best = max(best, (right - left) * height[left])
                left += 1
        return best


def main():
    s = Solution()
    print s.maxArea([0, 2])

if __name__ == '__main__':
    main()

File: java/src/Leetcode/ContainerWithMostWater.java
package Leetcode;

/**
 * Created by xin on 3/5/14.
 */
public class ContainerWithMostWater {
    public int maxArea(int[] height) {
        int l = 0, r = height.length - 1, best = 0;
        while (l < r) {
            best = Math.max(Math.min(height[l], height[r]) * (r - l), best);
            if (height[l] < height[r]) l++;
            else r--;
        }
        return best;
    }

    public static void main(String[] args) {
        ContainerWithMostWater s = new ContainerWithMostWater();
        int[] A = {1, 3, 1, 3, 1};
        System.out.println(s.maxArea(A));
    }
}
